packages/ui/__tests__/components/organisms/GroupByScrollList.test.js

```diff
@@ -0,0 +1,115 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import {shallow} from 'enzyme';
+import {GroupByScrollList, ScrollList, Text} from '@axelor/aos-mobile-ui';
+import {getGlobalStyles} from '../../tools';
+
+describe('GroupByScrollList', () => {
+  const data = [
+    {id: 1, name: 'Aa'},
+    {id: 2, name: 'Ab'},
+    {id: 3, name: 'Bb'},
+  ];
+  const renderItem = ({item}) => <Text>{item.name}</Text>;
+  const fetchData = jest.fn();
+  const fetchIndicator = jest.fn(currentItem => ({
+    title: currentItem.name[0].toUpperCase(),
+    numberItems: data.filter(item => item.name[0] === currentItem.name[0])
+      .length,
+    loading: false,
+  }));
+  const separatorCondition = (prevItem, currentItem) =>
+    prevItem.name[0] !== currentItem.name[0];
+
+  const props = {
+    loadingList: false,
+    data,
+    renderItem,
+    fetchData,
+    moreLoading: false,
+    isListEnd: false,
+    filter: false,
+    translator: key => key,
+    horizontal: false,
+    disabledRefresh: false,
+    separatorCondition,
+    fetchIndicator,
+  };
+
+  it('should render without crashing', () => {
+    const wrapper = shallow(<GroupByScrollList {...props} />);
+
+    expect(wrapper.exists()).toBe(true);
+  });
+
+  it('should render ScrollList with the correct props', () => {
+    const wrapper = shallow(<GroupByScrollList {...props} />);
+
+    expect(wrapper.find(ScrollList)).toHaveLength(1);
+    expect(wrapper.find(ScrollList).prop('loadingList')).toBe(
+      props.loadingList,
+    );
+    expect(wrapper.find(ScrollList).prop('data')).toEqual(props.data);
+    expect(wrapper.find(ScrollList).prop('fetchData')).toEqual(fetchData);
+    expect(wrapper.find(ScrollList).prop('moreLoading')).toBe(
+      props.moreLoading,
+    );
+    expect(wrapper.find(ScrollList).prop('isListEnd')).toBe(props.isListEnd);
+    expect(wrapper.find(ScrollList).prop('filter')).toBe(props.filter);
+    expect(wrapper.find(ScrollList).prop('translator')).toBe(props.translator);
+    expect(wrapper.find(ScrollList).prop('horizontal')).toBe(props.horizontal);
+    expect(wrapper.find(ScrollList).prop('disabledRefresh')).toBe(
+      props.disabledRefresh,
+    );
+  });
+
+  it('should render Separator at the beginning and when separatorCondition return true', () => {
+    const wrapper = shallow(<GroupByScrollList {...props} />);
+
+    for (let i = 0; i < data.length; i++) {
+      const renderItemElement = wrapper
+        .find(ScrollList)
+        .renderProp('renderItem')({
+        item: data[i],
+        index: i,
+      });
+      let prevItem = null;
+      if (i !== 0) {
+        prevItem = data[i - 1];
+      }
+      if (i === 0 || separatorCondition(prevItem, data[i])) {
+        expect(renderItemElement.find('Separator').length).toBe(1);
```

Comment:
It could be nice to check here that fetchIndicator was called two with the right argument.